### DISCUSSION TITLE
Can I use it to compile a ros package ?

### DISCUSSION THREAD

**Discussion by Karim-SUN** [pyinstaller/pyinstaller#8473] [Help]
Hi there, 
I am wondering if I could use pyinstaller to turn a pure python implementation of a ros1 package into an executable, and then have the nodes within the package run from the packaged executable when ros is runing?

**Comment by bwoodsend** [comment]
What would be the point? ROS depends on Python so you wouldn't be making your code any more portable.

**Reply by Karim-SUN** [reply]
It's that I need some code encryption to avoid exposing the src to other partners

**Reply by rokm** [reply]
You'll have to look for another tool, then. PyInstaller is not an obfuscation tool - it is trivial to obtain byte-compiled modules and entry-point script from the generated executable.

**Reply by Karim-SUN** [reply]
alright, thx anyway.😊
